src/stringSimilarity.js

```diff
@@ -1,17 +1,109 @@
 import escapeRegExp from 'lodash.escaperegexp';
 import matcher from 'matcher';
+import { isObject } from './helpers';
 
 export const WILDCARD_MARKER_ESCAPED = '{{\\*}}';
 export const WILDCARD_MARKER = '{{*}}';
+export const UNQUOTED_WILDCARD_PLACEHOLDER = '{{%}}';
+const searchUnquotedWildcards = new RegExp(/(([^:\s,]*)\{\{\*\}\}([^,:\s}\]]*))(?=[^,:}\]\s]*)/g);
+const searchModifiedWildcards = new RegExp(/(([^:\s,]*)\{\{%\}\}([^,\s}\]]*))(?=[^,}\]\s]*)/g);
 
-export default (source, target) => {
-  if (!source || (source || '') === (target || '')) {
-    return source === target;
-  }
-
+const getIsMatch = (source, target) => {
   const wildcardedSource = source
     .replace(new RegExp(escapeRegExp('*'), 'g'), '\\*')
     .replace(new RegExp(escapeRegExp(WILDCARD_MARKER_ESCAPED), 'g'), '*');
 
   return matcher.isMatch(target, wildcardedSource);
 };
+
+export const recursiveKeySort = (data) => {
+  if (data && isObject(data)) {
+    const sortedData = {};
+    Object.keys(data).sort().forEach((key) => {
+      sortedData[key] = recursiveKeySort(data[key]);
+    });
+    if (Array.isArray(data)) {
+      return Object.values(sortedData);
+    }
+    return sortedData;
+  }
+  return data;
+};
+
+export const replaceWildcards = (value, makeParsable) => {
+  let processedValue = value;
+  const regEx = makeParsable ? searchUnquotedWildcards : searchModifiedWildcards;
+  let resultArray = regEx.exec(processedValue);
+
+  while (resultArray !== null) {
+    const matchedString = resultArray[0];
+    const matchedStringIndex = regEx.lastIndex - matchedString.length;
+    let modifiedMatch;
+
+    if (makeParsable) {
+      const isWildcardInString = matchedString.startsWith('"')
+        && matchedString.endsWith('"');
+
+      if (!isWildcardInString) {
+        modifiedMatch = matchedString
+          .replace(WILDCARD_MARKER, UNQUOTED_WILDCARD_PLACEHOLDER)
+          .split('');
+        modifiedMatch.splice(0, 0, '"');
+        modifiedMatch.push('"');
+        modifiedMatch = modifiedMatch.join('');
+        const processedPortion = processedValue.slice(0, matchedStringIndex);
```

Comment:
Would it be simpiler to replace lines 48-53 with

        modifiedMatch = matchedString
          .replace(WILDCARD_MARKER, UNQUOTED_WILDCARD_PLACEHOLDER);
        modifiedMatch = '"'.concat(modifiedMatch.concat('"'));

Reply:
Thanks!

Reply:
I like it!  I've updated the logic, and done the same for the logic below that, which restores the original wildcard by removing the quotes around the match.